src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Runtime.Serialization.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using NBitcoin.RPC;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Features.BlockStore.Models;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.Utilities;
+using Stratis.Bitcoin.Utilities.JsonErrors;
+
+namespace Stratis.Bitcoin.Features.BlockStore.Controllers
+{
+    /// <summary>
+    /// Controller providing operations on a wallet.
+    /// </summary>
+    [Route("api/[controller]")]
+    public class BlockStoreController : Controller
+    {
+        private readonly IBlockStoreCache blockStoreCache;
+
+        /// <summary>Specification of the network the node runs on - regtest/testnet/mainnet.</summary>
+        private readonly Network network;
+
+        private readonly IConnectionManager connectionManager;
+
+        private readonly ConcurrentChain chain;
+
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        /// <summary>Provider of date time functionality.</summary>
+        private readonly IDateTimeProvider dateTimeProvider;
+
+        public BlockStoreController(
+            ILoggerFactory loggerFactory,
+            IBlockStoreCache blockStoreCache,
+            IConnectionManager connectionManager,
+            Network network,
+            ConcurrentChain chain,
+            IBroadcasterManager broadcasterManager,
+            IDateTimeProvider dateTimeProvider)
+        {
+            this.blockStoreCache = blockStoreCache;
```

Comment:
Are you using connectionManager, network, chain and dateTimeProvider?

Reply:
No I am not, sorry...